From 1c30725245c62baa7fcc9a83636264ea5453bde6 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 16 Apr 2019 18:30:37 -0700
Subject: [PATCH 0628/1852] v4l: xilinx: m2m: Fix buftype for s_selection ioctl

commit 161726e4398e7be9ba7851660c952ea0c032385b from
https://github.com/Xilinx/linux-xlnx.git

In current implementstion, driver is checking against _MPLANE buffer
types, but the type field in struct v4l2_selection is supposed to never
use the _MPLANE variants.

The commit eaec420f530d ("[media] v4l2-ioctl/exynos: fix G/S_SELECTION's
type handling") is converging non-mplane and _MPLANE formats to non-mpalne
formats. This is to allow applications to pass _MPLANE and non-mpalne
buffer types, and the driver only handles the non-mplane formats. Fix
this driver accordingly.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-m2m.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-m2m.c b/drivers/media/platform/xilinx/xilinx-m2m.c
index cbff079aef86..e9eb3b41973f 100644
--- a/drivers/media/platform/xilinx/xilinx-m2m.c
+++ b/drivers/media/platform/xilinx/xilinx-m2m.c
@@ -1053,8 +1053,8 @@ xvip_m2m_g_selection(struct file *file, void *fh, struct v4l2_selection *s)
 	struct xvip_m2m_dma *dma = ctx->xdev->dma;
 	int ret = 0;
 
-	if (s->type != V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE &&
-	    s->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE)
+	if (s->type != V4L2_BUF_TYPE_VIDEO_OUTPUT &&
+	    s->type != V4L2_BUF_TYPE_VIDEO_CAPTURE)
 		return -EINVAL;
 
 	switch (s->target) {
@@ -1082,8 +1082,8 @@ xvip_m2m_s_selection(struct file *file, void *fh, struct v4l2_selection *s)
 	u32 min_width, max_width;
 	int ret = 0;
 
-	if (s->type != V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE &&
-	    s->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE)
+	if (s->type != V4L2_BUF_TYPE_VIDEO_OUTPUT &&
+	    s->type != V4L2_BUF_TYPE_VIDEO_CAPTURE)
 		return -EINVAL;
 
 	switch (s->target) {
-- 
2.31.1

